confgenerator/fluentbit/modular.go

```diff
@@ -26,6 +26,67 @@ type Component struct {
 	Kind string
 	// Config is a set of key-value configuration pairs
 	Config map[string]string
+	// RepeatedConfig is used for configuration pairs where the
+	// key can appear in the output fluent bit config multiple times
+	RepeatedConfig map[string][]string
+}
+
+// ParserShared holds common parameters that are used by all processors that are implemented with fluentbit's "parser" filter.
```

Comment:
Can you please put this new type in a separate file since it doesn't interact with the rest of this file? (e.g. `processors.go`?

Out of curiosity, what was the reason this moved into a different package? The structure I was trying to establish was the the `confgenerator/fluentbit` package only contained non-user-visible code and structures, and all of the user-visible types with YAML tags etc. were in `confgenerator` or `apps`.

Reply:
I moved it intending to slide the functions that felt less user visible (generating the Component objects) into the fluentbit support package, but keep the Component function tightly associated with the shared parser struct. Additionally, the struct is intended to specify common parts that are very specific to Fluent Bit.

I'm happy to move it all back, or to move the struct back and keep the functions in fluentbit, or whatever makes sense.

Reply:
Moved into processors.go for now, but happy to make another change if you have a different thought @quentinmit